### ISSUE TITLE
Child products don't show images

### ISSUE THREAD

**Issue by SalahAdDin** [django-oscar/django-oscar#2099]
Hi guys,

I'm testing the current project and i found this:

![seleccion_197](https://cloud.githubusercontent.com/assets/5034215/17656409/730a590a-627f-11e6-8b39-3967b53ce85e.png)
"Chal rojo" is a child product and doesn't show any image.

![seleccion_196](https://cloud.githubusercontent.com/assets/5034215/17656410/730b8e1a-627f-11e6-9c29-bd5abfcf1ce0.png)
You can see here that "Chal" have a variant product called "Chal Rojo", "Chal" have image because we can add images to a parent product.

![seleccion_195](https://cloud.githubusercontent.com/assets/5034215/17656408/730970da-627f-11e6-8388-cf5ecbcbe372.png)
But, in the child product we can not add images, so, how can Oscar show the child product images if he hasn't image because Oscar doesn't leave to us add images to a child product?

**Comment by digicase** [comment]
Hi @SalahAdDin.

Before submitting an issue it would be best to search for already existing issues. A quick search gives these issues, where discussion has already has taken place.

https://github.com/django-oscar/django-oscar/issues?utf8=%E2%9C%93&q=child%20images

Please close this issue. Thanks.

If you need general help, instead of submitting issues it is best to use the discussion forum at https://groups.google.com/forum/#!forum/django-oscar

**Comment by digicase** [comment]
I would like to add that from personal experience Django Oscar is not an off the shelf solution if you need to manage a website that has products made up of parents with multiple child variations, such as a clothing store that sells products in different sizes and colours.

Oscar requires writing a lot of custom business logic in order to facilitate selling and managing these kinds of products well enough, including having to make significant changes to the Dashboard app for creating and managing products with variations. This includes adding support for child products to have their own images.

There are several open issues related to bringing support in Oscar to managing and selling these types of products, which you can use to garner more information on what Oscar does/does not currently support.

**Comment by SalahAdDin** [comment]
@digicase I'm sorry for say this but mailing list is, until now, useless, i do two questions and i haven't answers yet.

**Comment by digicase** [comment]
OK understood. But opening up redundant issues, and also leaving comments on open issues like you just did (which was simply a thumbs up emoji), is not in my opinion the best way to get help. The Oscar community is small, the core developers seem to be extremely busy and the original developers have all but left the project.

I've found the best way is to go ahead and write my own custom code for features I need, using Oscar as a base. I search the Google Group and the Gtihub issues for any pointers I can find.

**Comment by SalahAdDin** [comment]
Left the project? O no, why? :'(

I'm waiting for help in groups, when i'll finished my projects, or at least solved these problem i want contribute here.

Until now i forked a `django-oscar-payu` package but have problems with it.

**Comment by LennyLip** [comment]
I use simple hack :)

https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/catalogue/abstract_models.py#L625

```
except IndexError:
            if self.parent:  
                return self.parent.primary_image()
```

thats all, works for me.

**Comment by sasha0** [comment]
Fixed in 958a973d2282c94b569a7c00562f6f9b40ca5bb9.

**Comment by SalahAdDin** [comment]
@sasha0 Thank you man!
